docs/core/deploying/native-aot.md

```diff
@@ -0,0 +1,70 @@
+---
+title: NativeAOT deployment overview
+description: Learn what NativeAOT deployments are and why you should consider using it as part of the publishing your app with .NET 7 and later.
+author: lakshanf
+ms.author: lakshanf
+ms.date: 06/09/2022
+---
+# NativeAOT Deployment
+
+The .NET NativeAOT deployment model uses an optimized ahead-of-time (AOT) compiler that generates an application that contains native executable code. NativeAOT apps don't use the just-in-time (JIT) compiler when the application runs. Native AOT provides performance benefits (for example, startup time and working set improvements) and the ability to run .NET applications in restricted environments where a JIT is not allowed. NativeAOT applications are [self-contained applications](index.md#publish-self-contained) that target a specific runtime environment (RID), such as Linux x64 or Windows x64.
+
+There are some limitations in the .NET NativeAOT deployment model, with the main one being that no run-time code generation is possible. For more information, see [Impact of using the NativeAOT deployment](#impact-of-using-the-nativeaot-deployment). The support in the .NET 7 release is targeted towards console-type applications.
+
+> [!WARNING]
+> NativeAOT deployment is an experimental feature in .NET 7.
+
+## Publish NativeAOT - CLI
+
+Publish a NativeAOT application using the [dotnet publish](../tools/dotnet-publish.md) command.
+
+01. Add `<PublishAot>true</PublishAot>` to your project file.
+
+    This will produce a NativeAOT app and shows PublishAot compatibility warnings during build.
+
+    ```xml
+    <PropertyGroup>
+        <PublishAot>true</PublishAot>
+    </PropertyGroup>
+    ```
+
+01. Publish the app as for a specific runtime identifier using `dotnet publish -r <RID>`
+
+    The following example publishes the app for Windows as a NativeAOT application.
```

Comment:
This should talk about pre-requisites you need to install for the compilation.

(It should be simplified version of https://github.com/dotnet/runtime/blob/main/src/coreclr/nativeaot/docs/prerequisites.md.)